front/src/components/pages/GameListPage/GameListPage.tsx

```diff
@@ -56,15 +57,33 @@ export default function GameListPage(): React.ReactElement {
       return;
     }
     connect(userSocket);
-    userSocket.on('newMatch', (payload) => {
+    userSocket.on('newMatch', (payload: {
+      uuid:string, left:string, right:string
+    }) => {
       transferGameInfo(payload);
+      setPlayers([payload.left, payload.right]);
       setIsLoading(false);
       history.push('/pong');
     });
-    userSocket?.on('endMatch', () => {
+    userSocket.on('endMatch', () => {
       history.push('/game');
     });
-  }, [userSocket, history, check]);
+    history.listen(() => {
+      if (history.action === 'POP') {
+        if (players.includes(user.name)) {
+          userSocket.offAny();
+          userSocket.disconnect();
```

Comment:
```suggestion
          userSocket.disconnect();
```
disconnect 하면 offAny 할 필요가 있나요?

Reply:
disconnect시에 이벤트들이 모두 사라지는지 어떻게 되는지 몰라서 작성해두었는데 없애도 될 것 같습니다!